namerd/iface/interpreter-thrift/src/main/scala/io/buoyant/namerd/iface/ThriftNamerInterface.scala

```diff
@@ -100,14 +100,16 @@ object ThriftNamerInterface {
     protected[this] def nextStamp(): Stamp
 
     protected[this] final def update(v: Try[T]): Unit = {
-      val obs = Observation(nextStamp(), v)
-      val promise = synchronized {
-        val previous = pending
-        current = Some(obs)
-        pending = new Promise[Observation]
-        previous
+      if (!current.exists(_.value == v)) {
+        val obs = Observation(nextStamp(), v)
+        val promise = synchronized {
+          val previous = pending
+          current = Some(obs)
```

Comment:
Hmm.. maybe i'm misreading this.  Still the, read/write strikes me as racey.  I think we need to at least mark `current` as `@volatile` if it's being accessed outside of synchronized.  We should think about how this handles with concurrent updates, though...